newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -100,17 +104,30 @@ public List<NewsDto> getTopHeadlines() {
             savedTodayNewsDto = handleNewsSaving(isTodayInDb,futureArticles,today);
         }
 
-       return of(savedTodayNewsDto, savedYesterdayNewsDto)
-               .flatMap(Optional::stream)
-               .toList();
+       return List.of(savedTodayNewsDto,savedYesterdayNewsDto);
     }
 
-    private Optional<NewsDto> handleNewsSaving(boolean isInDb, List<NewsDto.Article> articles,LocalDate date) {
-        if(!isInDb && !articles.isEmpty()){
-           return Optional.of(saveNews(articles,date));
+    private NewsDto handleNewsSaving(boolean isInDb, List<NewsDto.Article> articles, LocalDate date) {
+
+        Optional<NewsEntity> existingNews = fetchNewsFromDatabase(isInDb, date);
+
+        return existingNews
+                .map(news -> {
+                    if (articles.size() > news.getTotalResults()) {
+                        return updateNews(articles, date);
+                    }
+                    return newsMapper.toDto(news);
+                })
+                .orElseGet(() -> saveNews(articles, date));
+    }
+
+    private Optional<NewsEntity> fetchNewsFromDatabase(boolean isInDb, LocalDate date) {
```

Comment:
Don't use entity in app layer try to use DTO

Reply:
Used DTO